sql/expression/arithmetic.go

```diff
@@ -346,14 +523,24 @@ func (a *Arithmetic) convertLeftRight(left interface{}, right interface{}) (inte
 	} else {
 		right, err = typ.Convert(right)
 		if err != nil {
-			return nil, nil, err
+			// TODO : any error here from converting should be added as warning
+			// the value is interpreted as 0, but we need to match the type of the other valid value
+			// to avoid additional conversion, the nil value is handled in each operation
+			right = nil
 		}
 	}
 
 	return left, right, nil
 }
 
 func plus(lval, rval interface{}) (interface{}, error) {
+	if lval == nil && rval == nil {
+		return 0, nil
+	} else if lval == nil {
+		return rval, nil
+	} else if rval == nil {
+		return lval, nil
+	}
```

Comment:
When does this code kick in? I had a hard time figuring it out... it seems like it's saying that `NULL+NULL` should eval to `0`, but that's not MySQL's behavior and it doesn't seem to be the way the code behaves when I test it locally either. Let me know if I'm just missing something silly.

Reply:
This is part of the last addition for fixing `select 'a' + 1` query, which MySQL allows, but Dolt throws error. We handle `select NULL + 1` as expected after evaluating the left and right values, but when we convert both the values into the same type, dolt throw error, `'a' + 1` is the case here where 'a' cannot be converted into number, so MySQL treats values that are not number as 0. This is an edge case for division operator that `select 'a'/1` returns `0` instead of `0.000`, which is why the nulls after conversion part is treated as 0s in each operation.

Reply:
Ahhh... that makes way more sense. Thanks for the explanation!